Content.Server/Interaction/InteractionSystem.cs

```diff
@@ -252,9 +254,10 @@ public override void DoAttack(EntityUid user, EntityCoordinates coordinates, boo
                         }
                     }
                 }
-                else if (!wideAttack && target != null && HasComp<ItemComponent>(target.Value))
+                else if (!wideAttack && target != null && HasComp<ItemComponent>(target.Value) && !_tagSystem.HasTag(target.Value, "NoPickupOnAttack"))
```

Comment:
are you sure a tag is the best way to do this rather than simply prioritizing hitting mobs over picking up items?

Reply:
My thought was, the tag solves this particular problem, but also adds the ability to exclude other Items from being picked up in combat if that is desired. 

You have a default behavior -- pick up all Items, even when in combat -- with a lightweight and flexible way of configuring that behavior agnostic of what other Types the target is composed of.

Is there a drawback to using a tag?

Reply:
Another use case I could think of, because I just worked on it, is opening a can of food. If you're empty handed and have combat turned on, you'll just pick up the can instead of bashing it open. Maybe this makes sense for some items, but certain destructible items that you might want to allow a player to bash open with their hands could just use NoPickupOnAttack.